### PR TITLE
refactored buttons

### PR DESCRIPTION
**Pull request by axelrod-blox**

### CODE REVIEW

File: src/app/components/applications/SSV/MyAccount/components/ClusterDashboard/ClusterDashboard.tsx [hunk 1]
```diff
@@ -143,12 +146,10 @@ const ClusterDashboard = () => {
...
           {rows.length > 0 && (
-            <Grid item className={`${classes.HeaderButton} ${classes.lightHeaderButton}`} onClick={moveToFeeRecipient}>
-              Fee Address
-              <Grid item className={classes.Pencil}/>
-            </Grid>
+            <SecondaryButton onClick={moveToFeeRecipient} text={'Fee Address'} size={ButtonSize.MD}/>
           )}
-          <Grid item className={classes.HeaderButton} onClick={moveToRegisterValidator}>Add Cluster</Grid>
+          <PrimaryButton onClick={moveToRegisterValidator} text={'Add Cluster'} size={ButtonSize.MD}/>
+          {/*<Grid item className={classes.HeaderButton} onClick={moveToRegisterValidator}>Add Cluster</Grid>*/}
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/ClusterDashboard/ClusterDashboard.tsx:152]: remove comment

File: src/app/components/applications/SSV/MyAccount/components/OperatorDashboard/OperatorDashboard.tsx [hunk 2]
```diff
@@ -126,32 +134,38 @@ const OperatorDashboard = () => {
   return (
     <Grid container className={classes.MyAccountWrapper}>
       <Grid container item className={classes.HeaderWrapper}>
-        <ToggleDashboards title={'Operators'} type={'operator'} />
-        <Grid container item xs className={classes.HeaderButtonsWrapper}>
-          <Grid item className={classes.HeaderButton} onClick={moveToRegisterOperator}>Add Operator</Grid>
-        </Grid>
+        <ToggleDashboards title={'Operators'} type={'operator'}/>
+        {/*<Grid container item xs className={classes.HeaderButtonsWrapper}>*/}
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/OperatorDashboard/OperatorDashboard.tsx:138]: remove comment

File: src/app/components/applications/SSV/MyAccount/components/Withdraw/ClusterFlow.tsx [hunk 3]
```diff
@@ -92,57 +106,71 @@ const ClusterFlow = ({ cluster, callbackAfterExecution, minimumLiquidationCollat
...
+      marginTop={0}
+      withoutNavigation
+      header={'Withdraw'}
+      body={secondBorderScreen}
+      bottom={[
+        <TermsAndConditionsCheckbox isChecked={isChecked} toggleIsChecked={() => setIsChecked(!isChecked)}
+                                    isMainnet={isMainnet}>
+          <div>
+          {showCheckBox && (
+            // <Grid item xs={12}>
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/Withdraw/ClusterFlow.tsx:157]: remove comments

File: src/app/components/applications/SSV/MyAccount/components/Withdraw/OperatorFlow.tsx [hunk 4]
```diff
@@ -11,9 +10,11 @@ import { getIsContractWallet, getIsMainnet } from '~app/redux/wallet.slice';
 import { useAppSelector } from '~app/hooks/redux.hook';
 import { IOperator } from '~app/model/operator.model';
 import { getOperatorBalance, withdrawRewards } from '~root/services/operator.service';
-import { useStores } from '~app/hooks/useStores';
+import PrimaryButton from '~app/atomics/PrimaryButton';
+import { ButtonSize } from '~app/enums/Button.enum';
 import { SingleOperator } from '~app/model/processes.model';
-import ProcessStore from '~app/common/stores/applications/SsvWeb/Process.store';
+import { useStores } from '~app/hooks/useStores';
+import ProcessStore from '../../../../../../common/stores/applications/SsvWeb/Process.store';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/Withdraw/OperatorFlow.tsx:17]: ~app/

File: src/app/components/applications/SSV/MyAccount/components/Withdraw/OperatorFlow.tsx [hunk 5]
```diff
@@ -60,48 +66,48 @@ const OperatorFlow = ({ operator }: { operator: IOperator }) => {
...
+          <Grid item container xs={6} className={classes.MaxButtonWrapper}>
+            <Grid item onClick={maxValue} className={classes.MaxButton}>
+              MAX
             </Grid>
-            {/* <Grid item xs={12} className={classes.BalanceInputDollar}> */}
-            {/*  ~$9485.67 */}
-            {/* </Grid> */}
+            <Grid item className={classes.MaxButtonText}>SSV</Grid>
           </Grid>
+          {/* <Grid item xs={12} className={classes.BalanceInputDollar}> */}
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/MyAccount/components/Withdraw/OperatorFlow.tsx:86]: remove comment

Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/Withdraw/OperatorFlow.tsx:86]: +

File: src/app/components/applications/SSV/MyAccount/components/Withdraw/OperatorFlow.tsx (hunk 4 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/Withdraw/OperatorFlow.tsx:17]: +

File: src/app/components/applications/SSV/MyAccount/components/Withdraw/ClusterFlow.tsx (hunk 3 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/Withdraw/ClusterFlow.tsx:157]: +

File: src/app/components/applications/SSV/MyAccount/components/OperatorDashboard/OperatorDashboard.tsx (hunk 2 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/OperatorDashboard/OperatorDashboard.tsx:138]: +

File: src/app/components/applications/SSV/MyAccount/components/ClusterDashboard/ClusterDashboard.tsx (hunk 1 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/ClusterDashboard/ClusterDashboard.tsx]: +

File: src/app/components/AllowanceButton.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+    margin-top: 16px;
+`;
+
+const ProgressStepsInner = styled.div`
+    display: flex;
+    flex-direction: row;
+`;
+
+
+const Step = styled.div<{ isCurrent: boolean, isFinished?: boolean }>`
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/AllowanceButton.tsx]: +

File: src/app/atomics/SecondaryButton.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  withoutBackgroundColor?: boolean
+  isDisabled?: boolean;
+}) => {
+
+  const handleOnClickFunction = () => !isDisabled && onClick();
+
+  return (
+    <Button reverseDirection={reverseDirection}  onClick={handleOnClickFunction} size={size} withoutBackgroundColor={withoutBackgroundColor} isDisabled={isDisabled}>
+      {text}
+      {icon && <Icon size={size}  path={icon}/>}
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/SecondaryButton.tsx]: +

File: src/app/atomics/SecondaryButton.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  reverseDirection?: boolean,
+  onClick: Function,
+  withoutBackgroundColor?: boolean
+  isDisabled?: boolean;
+}) => {
+
+  const handleOnClickFunction = () => !isDisabled && onClick();
+
+  return (
+    <Button reverseDirection={reverseDirection}  onClick={handleOnClickFunction} size={size} withoutBackgroundColor={withoutBackgroundColor} isDisabled={isDisabled}>
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/SecondaryButton.tsx]: +

File: src/app/atomics/SecondaryButton.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+const Icon = styled.div<{ path: string, size: ButtonSize }>`
+    width: 24px;
+    height: 24px;
+    background-size: contain;
+    background-position: center;
+    background-repeat: no-repeat;
+    background-image: ${({ path }) => `url(${path})`};
+`;
+
+const SecondaryButton = ({ size, text, icon, reverseDirection = false, onClick, withoutBackgroundColor = false, isDisabled = false }: {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/SecondaryButton.tsx]: +

File: src/app/atomics/SecondaryButton.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    justify-content: center;
+    padding: 0 18px;
+    align-items: center;
+    border-radius: 8px;
+    font-size: 16px;
+    font-weight: 600;
+    flex-direction: ${({ reverseDirection }) => reverseDirection ? 'row-reverse' : 'row'};
+    color: ${({ theme, isDisabled }) => isDisabled ? theme.colors.gray40 : theme.colors.primaryBlue};
+    cursor: pointer;
+    background-color: ${({ theme, withoutBackgroundColor, isDisabled }) => {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/SecondaryButton.tsx]: +

File: src/app/atomics/SecondaryButton.tsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import styled from 'styled-components';
+import { ButtonSize } from '~app/enums/Button.enum';
+
+const Button = styled.div<{ theme: any, size: ButtonSize, withoutBackgroundColor: boolean, isDisabled: boolean, reverseDirection: boolean }>`
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/SecondaryButton.tsx]: +

File: src/app/atomics/ErrorButton.tsx [hunk 12]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import styled from 'styled-components';
+import { ButtonSize } from '~app/enums/Button.enum';
+import Spinner from '~app/components/common/Spinner';
+
+const Button = styled.div<{ theme: any, size: ButtonSize, isDisabled: boolean, isLoading: boolean, reverseDirection: boolean }>`
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/ErrorButton.tsx]: +

File: src/app/atomics/ErrorButton.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+        if (size === ButtonSize.SM) {
+            return '40px';
+        }
+        if (size === ButtonSize.MD) {
+            return '48px';
+        }
+        return '60px';
+    }};
+    display: flex;
+    flex-direction: ${({ reverseDirection }) => reverseDirection ? 'row-reverse' : 'row'};
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/ErrorButton.tsx]: +

File: src/app/atomics/PrimaryButton.tsx [hunk 14]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/PrimaryButton.tsx]: +

File: src/app/atomics/PrimaryButton.tsx [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+const Icon = styled.div<{ path: string }>`
+    width: 24px;
+    height: 24px;
+    background-size: contain;
+    background-position: center;
+    background-repeat: no-repeat;
+    background-image: ${({ path }) => `url(${path})`};
+`;
+
+const PrimaryButton = ({ size, text, icon, reverseDirection = false, onClick, isDisabled = false, isLoading = false }: {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/PrimaryButton.tsx]: +

File: src/app/atomics/PrimaryButton.tsx [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+    padding: 0 18px;
+    font-weight: 600;
+    flex-direction: ${({ reverseDirection }) => reverseDirection ? 'row-reverse' : 'row'};
+    color: ${({ theme, isDisabled, isLoading }) => {
+        if (isLoading) {
+            return theme.colors.tint20;
+        }
+        return isDisabled ? theme.colors.gray40 : theme.colors.white;
+    }};
+    cursor: ${({ isDisabled }) => isDisabled ? 'default' : 'pointer'};;
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/PrimaryButton.tsx]: +

File: src/app/atomics/PrimaryButton.tsx [hunk 17]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import styled from 'styled-components';
+import { ButtonSize, IconDirection } from '~app/enums/Button.enum';
+import Spinner from '~app/components/common/Spinner';
+
+const Button = styled.div<{ theme: any, size: ButtonSize, isDisabled: boolean, isLoading: boolean, reverseDirection: boolean }>`
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/atomics/PrimaryButton.tsx]: +

**Pull request merged**